### PR TITLE
Add jtx cpp test for noripple flag (RIPD-1259):

### PR DESCRIPTION
**Pull request by willwray**
- Set and clear noripple flag
 - DefaultRipple on account
 - Set noripple on trustline with -ve balance
 - Pairwise noripple

### CODE REVIEW

File: src/test/rpc/NoRipple_test.cpp [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+public:
+    void
+    testSetAndClear()
+    {
+        testcase("Set and clear noripple");
+
+        using namespace jtx;
+        Env env(*this);
+
+        auto const gw = Account("gateway");
```
Comment (mellery451) [inline src/test/rpc/NoRipple_test.cpp]: minor: I think we've been preferring brace-style construction lately.

**Comment by mellery451** [comment]
👍

**Comment by vinniefalco** [comment]
Its perplexing that a branch which only adds test code can cause coverage to go down?

File: src/test/rpc/NoRipple_test.cpp (hunk 1 above)
Comment (willwray) [inline src/test/rpc/NoRipple_test.cpp]: me too. will fix.

**Comment by willwray** [comment]
I discussed the coverage decrease with Brad. We have no theories yet.

File: src/test/rpc/NoRipple_test.cpp [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+
+            // Check no-ripple flag on sender 'gateway'
+            auto lines = env.rpc("json", "account_lines", to_string(account_gw));
+            auto const& gline0 = lines[jss::result][jss::lines][0u];
+            BEAST_EXPECT(gline0[jss::no_ripple].asBool() == SetOrClear);
+
+            // Check no-ripple peer flag on destination 'alice'
+            lines = env.rpc("json", "account_lines", to_string(account_alice));
+            auto const& aline0 = lines[jss::result][jss::lines][0u];
+            BEAST_EXPECT(aline0[jss::no_ripple_peer].asBool() == SetOrClear);
```
Comment (nbougalis) [inline src/test/rpc/NoRipple_test.cpp]: I am confused... we set `no ripple` on one side of the trust line, but we look for it in both?

Comment (seelabs) [inline src/test/rpc/NoRipple_test.cpp]: @nbougalis It looks like one check is for `no_ripple`, and the other is for `no_ripple_peer`. `no_ripple` is for this line, `no_ripple_peer` is if the peer account has enabled the flag.

Comment (nbougalis) [inline src/test/rpc/NoRipple_test.cpp]: Yeah, I'm fine with this. I didn't notice the `_peer` suffix.

**Comment by nbougalis** [comment]
:+1:

File: src/test/rpc/NoRipple_test.cpp [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+        env.fund(XRP(10000), gw, alice);
+
+        auto const USD = gw["USD"];
+
+        Json::Value account_gw;
+        account_gw[jss::account] = gw.human();
+        Json::Value account_alice;
+        account_alice[jss::account] = alice.human();
+
+        for (auto SetOrClear : {true,false})
```
Comment (seelabs) [inline src/test/rpc/NoRipple_test.cpp]: Nit: You could change this so it loops over `{tfSetNoRipple, tfClearNoRipple}`. That gets rid of the ternary `if` statement below. Also `SetOrClear` should not be capitalized.

Comment (nbougalis) [inline src/test/rpc/NoRipple_test.cpp]: Good idea.

File: src/test/rpc/NoRipple_test.cpp [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+        Env env(*this);
+
+        auto const gw = Account("gateway");
+        auto const alice = Account("alice");
+
+        env.fund(XRP(10000), gw, alice);
+
+        auto const USD = gw["USD"];
+
+        Json::Value account_gw;
```
Comment (seelabs) [inline src/test/rpc/NoRipple_test.cpp]: Style nit: `camelCase` not `snake_case`

Comment (vinniefalco) [inline src/test/rpc/NoRipple_test.cpp]: I would have gone with `jgw` but that's just a cosmetic issue

File: src/test/rpc/NoRipple_test.cpp (hunk 3 above)
Comment (willwray) [inline src/test/rpc/NoRipple_test.cpp]: camelCased but left it Boolean as it's also used to test the flag via asBool()

File: src/test/rpc/NoRipple_test.cpp (hunk 4 above)
Comment (willwray) [inline src/test/rpc/NoRipple_test.cpp]: Went with gwJson

File: src/test/rpc/NoRipple_test.cpp [hunk 5]
```diff
@@ -69,7 +69,7 @@ class NoRipple_test : public beast::unit_test::suite
     void
     testNegativeBalance()
     {
-        testcase{"Set noripple on a line with negative balance"};
+        testcase("Set noripple on a line with negative balance");
```
Comment (vinniefalco) [inline src/test/rpc/NoRipple_test.cpp]: Good change. calls to `testcase()` are supposed to look like function calls.

**Comment by nbougalis** [comment]
682b1638f5331a157033ecfa223baac912c7fa88 was merged as 5b09dc731fd578d697dc703852a268df0f1cbb41.

43fe7a9dbf497d7323572eabf31f80702c072c03 and d8b0ccb6ad13a9b2a26cca26fc078e7804e934d3 will be submitted separately.

**Pull request closed without merging**